### PR TITLE
Es6 behavior

### PR DESCRIPTION
**Pull request by jstefaniuk-d2l**

### CODE REVIEW

File: behaviors/d2l-upcoming-assessments-behavior.js [hunk 1]
```diff
@@ -125,81 +124,66 @@ var upcomingAssessmentsBehaviorImpl = {
...
-		var requests = [];
+		const overdueUserUsages = this._concatActivityUsageTypes(overdueUserActivityUsages);
+		const supportedUserUsages = this._concatActivityUsageTypes(userActivityUsages);
 
-		var overdueUserUsages = this._concatActivityUsageTypes(overdueUserActivityUsages);
-		var supportedUserUsages = this._concatActivityUsageTypes(userActivityUsages);
+		const requests = supportedUserUsages.map(async(userActivityUsage) => {
+			const organizationRequest = this._getOrganizationRequest.call(this, userActivityUsage, getToken, userUrl, abortSignal);
+			const activityRequest = this._getActivityRequest.call(this, userActivityUsage, getToken, userUrl);
+			const userActivityUsageHref = userActivityUsage.getLinkByRel('self').href;
```
Comment (EvanLemieux) [inline behaviors/d2l-upcoming-assessments-behavior.js:133]: This is a lot cleaner. I like this.

File: behaviors/d2l-upcoming-assessments-behavior.js [hunk 2]
```diff
@@ -217,23 +201,22 @@ var upcomingAssessmentsBehaviorImpl = {
 	},
 
 	_getCustomRangeAction: function(activitiesEntity, dateObj) {
-		var self = this;
```
Comment (EvanLemieux) [inline behaviors/d2l-upcoming-assessments-behavior.js:220]: Was this ever a recommended pattern? This seems so foreign to me lol

Comment (jstefaniuk-d2l) [reply behaviors/d2l-upcoming-assessments-behavior.js:220]: Yeah, back before `() => {}` was a thing, the `this` in anonymous functions was different from the context you created them in, so you would either have to `.bind(this)` them or define a `self` var for referencing instead

Comment (EvanLemieux) [reply behaviors/d2l-upcoming-assessments-behavior.js:220]: Ohhh yeah that makes a lot of sense. I know JS devs always joked about not knowing what `this` is, but it really didn't seem that bad. I only got into it after ES6 though so I guess the before times were even worse lol

**Review by EvanLemieux — APPROVED** [review]
This all seems pretty legit. LGTM :+1:

**Comment by jstefaniuk-d2l** [comment]
I may hold off on merging this since it may be more of a "transition" step to something newer

**Pull request merged**